apps/backend/src/db/migrations/20250606_init.ts

```diff
@@ -0,0 +1,304 @@
+import { type CreateTableBuilder, Kysely, sql } from 'kysely';
+
+const addDefaultColumns = (ctb: CreateTableBuilder<any, any>) => {
+  return ctb
+    .addColumn('id', 'uuid', (col) => col.primaryKey().defaultTo(sql`gen_random_uuid()`))
+    .addColumn('createdAt', 'date', (col) => col.notNull().defaultTo(sql`CURRENT_TIMESTAMP`))
+    .addColumn('modifiedAt', 'date', (col) => col.notNull().defaultTo(sql`CURRENT_TIMESTAMP`)); // I don't know how to set this to update automatically
+};
+
+const addNameCheckConstraint = (ctb: CreateTableBuilder<any, any>) => {
+  return ctb.addCheckConstraint('valid_name', sql`"name" ~ '^[a-zA-Z0-9 .,_:;!?()/\-]{1,256}$'`);
+};
+
+// --- Table Creation Helper Functions ---
+async function createUserTable(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .createTable('User')
+    .$call(addDefaultColumns)
+    .addColumn('email', 'varchar(256)', (col) => col.notNull().unique())
+    .addColumn('passwordHash', 'varchar(256)', (col) => col.notNull())
+    .addColumn('verified', 'boolean', (col) => col.notNull().defaultTo(false))
+    .addColumn('emailVerificationTokenHash', 'varchar(64)')
+    .addColumn('emailVerificationTokenExpiresAt', 'timestamptz(6)')
+    .addColumn('passwordResetTokenHash', 'varchar(64)')
+    .addColumn('passwordResetTokenExpiresAt', 'timestamptz(6)')
+    .addColumn('refreshTokenHash', 'varchar(64)')
+    .addColumn('refreshTokenExpiresAt', 'timestamptz(6)')
+    .addCheckConstraint('modified_after_created', sql`"modifiedAt" >= "createdAt"`)
```

Comment:
Can't this also be a dedicated function and added with `$call`? This line is repeated very often.